src/views/tutorial-view/components/image-config/index.tsx

```diff
@@ -0,0 +1,133 @@
+import { Children, isValidElement, ReactElement, ReactNode } from 'react'
+import MdxImg from '../mdx-img'
+// import classNames from 'classnames'
+import s from './image-config.module.css'
+
+/**
+ * Configure extra settings for images via MDX.
+ * Designed to wrap a single markdown image.
+ */
+export default function ImageConfig({
+  /** Hide borders (temporarily configurable while we remove baked-in borders from existing images) */
+  hideBorder = false,
+  /** A caption to display below the image. */
+  caption,
+  /** An MDX `img`, e.g. `![some alt text](path/to/img.jpg)` */
+  children,
+}: {
+  caption?: string
+  hideBorder?: boolean
+  children: ReactNode
+}): ReactElement {
+  //  emit a warning if no meaningful props are present
+  if (!hideBorder && !caption) {
+    console.warn(
+      `Warning: <ImageConfig /> was initialized with default props. Please remove <ImageConfig /> if configuration is not needed.`
+    )
+  }
+
+  // TODO: fully deprecate the `hideBorder` prop, then remove this warning
+  // Task: https://app.asana.com/0/0/1200167326263350/f
+  if (hideBorder) {
+    console.warn(
+      `Warning: <ImageConfig /> was initialized with a deprecated prop "hideBorder". This property will be deprecated in the near future. If you've provided an image caption, a border will be displayed even if "hideBorder" is set. If your image has a border built into it, please crop it to remove it.`
+    )
+  }
+
+  const imgElement = getValidatedImgChild(children)
+  // Validate that the { src, alt, title } props are present
+  const { src, alt, title } = imgElement.props
+  // Validate that src is present, and is a non-empty string
+  if (typeof src !== 'string' || src == '') {
+    throw new Error(
+      `Images in MDX must have an source defined. Use the format ![some alt text](path/to/img.jpg). Image details: ${JSON.stringify(
+        { src, alt, title }
+      )} `
+    )
+  }
+
+  // TODO: remove this once we're done reviewing
+  // (here for now because otherwise, previewing this seemed difficult)
+  const TEMP_DEV_CAPTION =
```

Comment:
Going to remove this before merge; I needed some way to test the caption prop so hooked into a specific image source (on [this page](https://dev-portal-git-zsadd-tutorial-img-component-hashicorp.vercel.app/vault/tutorials/getting-started-ui/getting-started-intro-ui)).

Related: we could, I think, _not_ implement the caption prop for now, since authors can't really use it yet. Or we could implement it now, and also backport these changes to the `learn` repo so that authors can start using `caption` (and the new component look & feel, apart from the system fonts) right away 👍

Reply:
I think the caption prop is a great addition and we should leave it in. Will save us time down the road. 😊

Reply:
Authors are planning on making dev dot content changes on a separate branch in tutorials so they could use this feature there? Or yes we could backport to Learn! Either way.

Reply:
☝️ @kendallstrautman If we want to backport to `learn` - what's the best board on which to surface that as a task? I think it could be cool to do if authors want it, though I imagine it might be lower on the list than other priorities. (Random thought, but if we do want to backport, I think we could wait till we have the docs use case accounted for as well in our `dev-dot` `ImageConfig`, and _then_ do the backporting).